File: one-to-one-videocall/src/main/resources/static/scripts/model/PeerMessage.js
import PeerMessageCode from './PeerMessageCode.js';

class PeerMessage {

    /**
     * @param {{code: PeerMessageCode, jsonPayload: String, senderUserId: Number}} properties
     */
    static fromProperties(properties) {
        return new PeerMessage(properties.code, properties.jsonPayload, properties.senderUserId);
    }

    /**
     * @param {PeerMessageCode} code
     * @param {String} jsonPayload
     * @param {Number} senderUserId
     */
    constructor(code, jsonPayload, senderUserId) {
        this.code = code;
        this.jsonPayload = jsonPayload;
        this.senderUserId = senderUserId;
    }

}

export default PeerMessage;

File: one-to-one-videocall/src/main/java/fr/marcworld/webrtcdemo/dtos/PeerMessage.java
package fr.marcworld.webrtcdemo.dtos;

public class PeerMessage {

    private PeerMessageCode code;
    private String jsonPayload;
    private int senderUserId;

    public PeerMessage() {
    }

    public PeerMessage(PeerMessageCode code, String jsonPayload, int senderUserId) {
        this.code = code;
        this.jsonPayload = jsonPayload;
        this.senderUserId = senderUserId;
    }

    public PeerMessageCode getCode() {
        return code;
    }

    public void setCode(PeerMessageCode code) {
        this.code = code;
    }

    public String getJsonPayload() {
        return jsonPayload;
    }

    public void setJsonPayload(String jsonPayload) {
        this.jsonPayload = jsonPayload;
    }

    public int getSenderUserId() {
        return senderUserId;
    }

    public void setSenderUserId(int senderUserId) {
        this.senderUserId = senderUserId;
    }

    @Override
    public String toString() {
        return "PeerMessage{" +
                "code=" + code +
                ", jsonPayload='" + jsonPayload + '\'' +
                ", senderUserId=" + senderUserId +
                '}';
    }
}
